api/proof.js

```diff
@@ -13,15 +13,14 @@ var app = express();
 app.get('/:trackingId', async (req, res) => {
 
   req.checkParams('trackingId', 'Invalid trackingId').notEmpty();
-  req.checkQuery('userId', 'Invalid userId').notEmpty();
 
   var errors = await req.getValidationResult();
   if (!errors.isEmpty()) {
     return res.status(HttpStatus.BAD_REQUEST).json({ error: `there have been validation errors: ${util.inspect(errors.array())}` });
   }
 
   var trackingId = decodeURIComponent(req.params.trackingId);
-  var userId = decodeURIComponent(req.query.userId);
+  var userId = req.headers['user-id'];
```

Comment:
just add a validation that userId is not empty

if (!userId) {
      return res.status(HttpStatus.BAD_REQUEST).json({ error: `userId was not found in headers` });
    }

Reply:
done!